### ISSUE TITLE
Re-use HTTP connections

### ISSUE THREAD

**Issue by jomo** [crafatar/crafatar#133]
We should look into using [`request`](https://github.com/request/request)'s agent options (`forever`?) option, and use [`http.Agent`](https://iojs.org/api/http.html#http_class_http_agent).

That should allow us to re-use established TCP sessions which is faster and reduces the number of open sockets.

**Comment by kylerschin** [comment]
yes we should... i am also working on this along with TCP and HTTPS support.

**Comment by kylerschin** [comment]
thats more of the host problem....
